iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ThreeWindingsTransformerXml.java

```diff
@@ -154,8 +188,44 @@ protected void readSubElements(ThreeWindingsTransformer tx, NetworkXmlReaderCont
                     break;
 
                 default:
-                    super.readSubElements(tx, context);
+                    readVersionedSubElements(context.getReader().getLocalName(), tx, context, (t, c) -> {
+                        try {
+                            super.readSubElements(tx, context);
+                        } catch (XMLStreamException e) {
+                            throw new UncheckedXmlStreamException(e);
+                        }
+                    });
             }
         });
     }
+
+    private static void readVersionedSubElements(String localName, ThreeWindingsTransformer tx, NetworkXmlReaderContext context,
+                                                 BiConsumer<ThreeWindingsTransformer, NetworkXmlReaderContext> consumer) throws XMLStreamException {
+        switch (context.getVersion()) {
+            case V_1_1:
+                switch (localName) {
+                    case "ratioTapChanger1":
+                        readRatioTapChanger(1, tx.getLeg1(), context);
+                        break;
+                    case "phaseTapChanger1":
+                        readPhaseTapChanger(1, tx.getLeg1(), context);
+                        break;
+                    case "phaseTapChanger2":
+                        readPhaseTapChanger(2, tx.getLeg2(), context);
+                        break;
+                    case "phaseTapChanger3":
+                        readPhaseTapChanger(3, tx.getLeg3(), context);
+                        break;
+                    default:
+                        consumer.accept(tx, context);
+                        break;
+                }
+                break;
+            case V_1_0:
+                consumer.accept(tx, context);
+                break;
+            default:
+                throw new PowsyblException("XIIDM version " + context.getVersion().toString(".") + " is not supported for ThreeWindingsTransformer.");
+        }
+    }
```

Comment:
I'm not sure about the code design...To avoid code duplication, we could introduce a function that use a predicate to know if the attribute should be read (or write)? We can also use readOptionalDoubleAttribute for new attributes to reduce the complexity of the code...

I think we have to create a design, developer could follow for the new changes in the format. I'm not sure switch/case is the best design.

The default case should not throw PowsyblException but AssertionError: it should not happen?!

Reply:
I agree with you proposition, I was not so sure myself about the way I coded it.

Actually, after some thinking, I think the default case should be the same as the most recent version (the serialization/reading of an equipment does not change for more recent version of IIDM unless explicitly specified), what do you think?